mlsdbtool/mcc_update.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# This will take the mcc data from the file "data/MCC" in the repository
+# tzdata-timed ( https://github.com/sailfishos/tzdata-timed ) and use it
+# to update the code in this repository.
+
+declare -i mcc
+HEADER="../common/mccmapping.h"
+SCRIPT="./mccmapping"
+mcc_file=$1
+if [ -z "$mcc_file" ] ; then
+    echo "Usage: $0 [MCC file]" >&2
+    exit 1
+fi
+if [ ! -e $HEADER ] ; then
+    echo "ERROR: Unable to locate header $HEADER" >&2
+    exit 1
+fi
+if [ ! -e $SCRIPT ] ; then
+    echo "ERROR: Unable to locate script file $SCRIPT" >&2
+    exit 1
+fi
+
+mccs=`cat $mcc_file | grep -v ^# | cut -d ' ' -f1 | sort -nu | tr '\n' ',' | sed 's/^,//;s/,$/,1024/'`
+count=`echo -n $mccs | tr -dc ',' | wc -c`
+
+if [ $count -gt 255 ] ; then
+    echo -e "We have a problem. The functionality of mlsdbtools assumes there are fewer than\n256 unique mccs. The file provided gives us $count. Some manual intervention is needed." >&2
+    exit 1
+fi
+
+echo "Updating $HEADER"
+sed -i "s/=.*/= \{$mccs\}\;/;s/... unique/$count unique/" $HEADER
+
+echo "Updating $SCRIPT"
+last_mcc=0
+first=0
+echo "# MCC -> country code mapping from tzdata-timed repository data/MCC" > $SCRIPT
+for element in `cat $mcc_file | grep -v ^# | cut -d ' ' -f1,3 | sort -u` ; do
```

Comment:
Quoting/escaping?
```
for element in `cat $mcc_file | grep -v ^# | cut -d ' ' -f1,3 | sort -u` ; do
               ^-- SC2013: To read lines rather than words, pipe/redirect to a 'while read' loop.
               ^-- SC2006: Use $(...) notation instead of legacy backticked `...`.
                    ^-------^ SC2086: Double quote to prevent globbing and word splitting.
                    ^-------^ SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.

Did you mean: 
for element in $(cat "$mcc_file" | grep -v ^# | cut -d ' ' -f1,3 | sort -u) ; do

Reply:
After looking this again I think might it be better to
`grep -v ^# "$mcc_file"| cut -d ' ' -f1,3 | sort -u| while read -r element ; do` 
?